spec/services/mixpanel_service_spec.rb

```diff
@@ -919,5 +920,53 @@ def inst_test
         )
       )
     end
+
+    context "when dropping events" do
+      before do
+        routes.draw { get "index" => "anonymous#index" }
+      end
+
+      context "for those not allowed" do
+        def self.test_security_metrics_ips
+          security_metrics_subnet = IPAddr.new("192.211.152.0/24") # 255 IP addresses in a Class C subnet
+          ip_addresses = security_metrics_subnet.to_range
+          ip_addresses.each { |ip_address|
+            it "drops those from the IP address #{ip_address}" do
+              @request.remote_addr = ip_address.to_string
+
+              get :index
+
+              expect(fake_tracker).not_to have_received(:track)
+            end
+          }
+        end
+
+        context "for SecurityMetrics" do
+          self.test_security_metrics_ips()
+        end
+
+        it "drops events coming from non-public AWS domains" do
+          request.set_header("HTTP_HOST", "ec2-18-204-251-64.compute-1.amazonaws.com")
+
+          get :index
+
+          expect(fake_tracker).not_to have_received(:track)
+        end
+
+        it "drops events coming from status checks" do
+          get :index, params: { source: :hund }
```

Comment:
I think in reality the source would come in as a string rather than a symbol so the test should probably use a string